operator/internal/status/storage.go

```diff
@@ -25,5 +42,47 @@ func SetStorageSchemaStatus(ctx context.Context, k k8s.Client, req ctrl.Request,
 		Schemas: schemas,
 	}
 
+	for _, sc := range schemas {
+		date, err := sc.EffectiveDate.UTCTime()
+		if err != nil {
+			return kverrors.Wrap(err, "failed to parse effective date")
+		}
+		if sc.Version == lokiv1.ObjectStorageSchemaV13 && date.Before(cutoff) {
+			schemaVersionMap[sc.Version] = true
+			continue
+		} else {
+			schemaVersionMap[sc.Version] = false
+			oldSchemas = append(oldSchemas, sc)
+		}
+	}
+
+	if len(oldSchemas) > 0 {
+		if schemaVersionMap[lokiv1.ObjectStorageSchemaV13] {
+			//s.Status.Storage.Schemas[] = StorageSchemaOutOfRetention
+			if err := updateSchemaStatus(s.Status.Storage.Schemas, StorageSchemaOutOfRetention); err != nil {
+				return kverrors.Wrap(err, "error updating schema status")
+			}
+		} else {
+			//s.Status.Storage.SchemaStatus = StorageSchemaNeedsUpgrade
+			if err := updateSchemaStatus(s.Status.Storage.Schemas, StorageSchemaNeedsUpgrade); err != nil {
```

Comment:
I don't know if I would "attach" the "needs upgrade" warning to a schema as this might be interpreted as "oh, I need to change this version to `v13` instead of "you need to add a new schema in the future which uses v13".

Reply:
it's called `StorageSchemaNeedsUpgrade` but the text is actually `"Consider upgrading to schema V13 to use TSDB shipper"`
so it's more a recommendation than a requirement

Reply:
That's true, but I still think attaching this text to an existing schema is not what we want, because the user can not upgrade an existing schema, they have to add a new one that gets activated in the future.